### PR TITLE
BC-224 - [Task] - Student - Filter the task and submit it

### PR DESCRIPTION
**Pull request by Uzaeir**
# Description

 Adds the test case for new task page based on nuxt client for the student. 

In this test case, student login with their credentials and see the open task in the new task menu, then student will submit the task and naviagate back to the new task menu to see the updated status of the task i.e. from open to complete task.

## Links to Tickets or other pull requests

- https://ticketsystem.hpi-schul-cloud.org/browse/BC-224

## Changes

  This PR adds the test case for new nuxt client based task page for student role.

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.hpi-schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: features/tasks/nuxtSubmitTask.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@task @nuxtTask @nuxtSubmitandCompleteTask
+Feature: View the task in new nuxt task menu and submit it
+
+    Background: User submit the opened tasks and view the tasks in completed tab
+        Given user arrives on the Schul-Cloud homepage
+
+    @nuxtSubmitAndViewTask @e2eCore
```
Comment (d3n153) [inline features/tasks/nuxtSubmitTask.feature:7]: You can remove the @e2eCore.

File: features/tasks/nuxtSubmitTask.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        When <userRole> logs in
+        And <userRole> clicks the current task tab
+        And <userRole> clicks on without duedate task section
+        And <userRole> clicks task
+        And <userRole> clicks on Submission tab
+        And <userRole> sets submission text 'Test submission text'
+        And <userRole> clicks on submit button
+        And <userRole> clicks the current task tab
+        And <userRole> clicks completed task tab
+        Then <userRole> clicks at ungraded task
```
Comment (d3n153) [inline features/tasks/nuxtSubmitTask.feature:18]: Rephrase the final step so it's an actual check for something. For example 'Then <userRole> can see ungraded task with name '...' <- what ever the name of the task is.

File: page-objects/pages/TASKListPage.js [hunk 3]
```diff
@@ -136,6 +141,15 @@ async function clickDeleteTaskButtonInPopup() {
 	await waitHelpers.waitUntilAjaxIsFinished();
 }
 
+async function clickTaskWithoutDuedate(){
+	await elementHelpers.clickAndWait(clickWithoutDueDate);
+}
+
+async function clickUngradedTask(){
+	await elementHelpers.clickAndWait(ungradedTaskSection);
+	await elementHelpers.clickAndWait(ungradedTask);
```
Comment (d3n153) [inline page-objects/pages/TASKListPage.js:150]: I'm thinking of dividing this into two seperate steps. One for going to the ungraded tasks section and one for actually opening the ungraded task.

File: page-objects/pages/TASKPage.js [hunk 4]
```diff
@@ -14,13 +14,20 @@ const remoteFilePathInput = 'input[type=file][class=dz-hidden-input]';
...
 const commentGradingTabSel = '#feedback-tab-link';
 const hometasksTabSel = 'button[data-testid="hometasks"]';
+const nuxtHometasksTabSel = '[data-testid="Offene Aufgaben"] > .link-name';
 const taskRatingInput = '[data-testid="evaluation_procent"]';
 const ratingViewSel = '.grade';
 const remarkViewSel = '.ckcontent.comment';
 const submissionsTable = '#submissions table';
 const submissionRow = `${submissionsTable} tbody tr.userinfo`;
+const withoutDueDateTask = '.v-expansion-panel--next-active > .v-expansion-panel-header';
+const completedTaskTab = '[aria-selected="false"]';
```
Comment (d3n153) [inline page-objects/pages/TASKPage.js:24]: const withoutDueDateTask and const completedTaskTab are bad selectors.
Ask your team, or we can try it ourselfs, to create data-testIds for those panels.

File: page-objects/pages/TASKListPage.js [hunk 5]
```diff
@@ -16,6 +18,9 @@ const taskTitleContainer = '.assignment.card .title';
 const taskDescriptionContainer = '.assignment .text-muted.ckcontent';
 const taskContainer = '.homework li.card';
 const deleteTaskButtonInPopup = '.delete-modal button.btn-submit';
+const clickWithoutDueDate = '[href="/homework/5fa3b9bea9c31a26f4d1da93"]';
+const ungradedTaskSection = '.v-window-item--active > section > .v-item-group > [aria-expanded="false"] > .v-expansion-panel-header';
+const ungradedTask = '.v-list-item__title';
```
Comment (d3n153) [inline page-objects/pages/TASKListPage.js:24]: const ungradedTaskSection and const ungradedTask are bad selectors.
Ask your team, or we can try it ourselfs, to create data-testIds for those panels.

**Review by d3n153 — CHANGES_REQUESTED** [review]

File: page-objects/pages/TASKListPage.js [hunk 6]
```diff
@@ -16,6 +18,9 @@ const taskTitleContainer = '.assignment.card .title';
 const taskDescriptionContainer = '.assignment .text-muted.ckcontent';
 const taskContainer = '.homework li.card';
 const deleteTaskButtonInPopup = '.delete-modal button.btn-submit';
+const clickWithoutDueDate = '[href="/homework/5fa3b9bea9c31a26f4d1da93"]';
```
[the remark this answers is not in the document]
Comment (yuliashikhareva) [reply page-objects/pages/TASKListPage.js:21]: I would say that we don't need to define  { default: $ } and  { expect } as they are globally pre-defined

File: page-objects/pages/TASKListPage.js (hunk 3 above)
Comment (yuliashikhareva) [reply page-objects/pages/TASKListPage.js:150]: Agree. Please split it up to functions like clickUngradedTaskSection and clickUngradedTask. You will need two function calls, one after another, in ..-steps.js file

File: page-objects/pages/TASKPage.js (hunk 4 above)
Comment (yuliashikhareva) [reply page-objects/pages/TASKPage.js:24]: Absolutely agree about the selectors. Ideally we have to put data-testids or at least take those selectors which are unique.

**Review by d3n153 — COMMENTED** [review]
Since most of the comments are resolved and I'll be on holiday next week I'll change my 'requested changes' review to 'comment'. Otherwise you might get problems merging it.

**Pull request closed without merging**